### ISSUE TITLE
Remove the filename from the class name without babel

### ISSUE THREAD

**Issue by slava-lu** [styled-components/styled-components#3373]
**Feature request**

It looks like the generated class name for the styled component includes the filename, unless you mangle with the babel plugin and change it for the whole app.  It would be though nice to somehow override it for a specific component without going to app's wide configuration.

The use case: I need to create a honeypot form component to protect my form from Spammers. I want the file to be called `honeypot.js` so that other developers could understand what it does, but I do not want the word `honeypot` to appear anywhere in the code so that scripts are not be able to find the trick. 

Basically I need to get rid of the word on the screenshot while keeping the filename `honeypot.js`

![image](https://user-images.githubusercontent.com/32396113/104208045-c8dca980-5430-11eb-82ab-7e1147f29206.png)

**Comment by kitten** [comment]
I believe you're looking for the following potentially? https://styled-components.com/docs/tooling#control-the-components-displayname

If not though, this may be an issue you'd want on https://github.com/styled-components/babel-plugin-styled-components instead.
